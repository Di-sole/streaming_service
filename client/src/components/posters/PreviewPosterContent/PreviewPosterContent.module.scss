.container {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(0, 0, 0, 0.75);
    opacity: 0;
    transition: 0.3s;

    &__icons {
        position: absolute;
        right: 8px;
        top: 8px;
        display: flex;
        flex-direction: column;
        align-items: center;
    }

    &__icon {
        width: 25px;
        height: 20px;

        &_favorite {
            margin-bottom: 4px;
        }

        &_similar {
            height: 25px;
            margin-bottom: 4px;
        }

        &_rate {
            width: 20px;
            margin-bottom: 8px;
        }

        &_dislike {
            width: 18px;
            height: 18px;
        }
    }

    &__info {
        position: absolute;
        left: 16px;
        right: 16px;
        bottom: 16px;
        /* font-size: 13px;
        line-height: 16px;
        font-weight: 500;
        color: #d9d7e0;*/
    }

    /* &__rating {
        margin-bottom: 12px;
        color: #fff;
        &_integer {
            font-size: 24px;
        }

        &_faction {
            font-size: 20px;
        }
    }*/

    &__text p {
        margin-top: 4px;
        display: -webkit-box;
        overflow: hidden;
        -webkit-line-clamp: 2;
        -webkit-box-orient: vertical;

        &:first-child {
            text-transform: capitalize;
        }
    }

    &:hover {
        opacity: 1;
    }
}
