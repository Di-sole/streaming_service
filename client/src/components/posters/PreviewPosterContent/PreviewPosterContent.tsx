import IMovies from '@/models/IMovies';
import Image from 'next/image';
import bookmark from '../../../../public/icons/posters/bookmark.png';
import wand from '../../../../public/icons/posters/wand.png';
import star from '../../../../public/icons/posters/star.png';
import circle from '../../../../public/icons/posters/circle.png';
import styles from './PreviewPosterContent.module.scss';
import PreviewPosterContentBrief from '../PreviewPosterContentBrief/PreviewPosterContentBrief';
import IMovie from '@/models/IMovie';

interface PreviewPosterContentProps {
    movie: IMovies | IMovie;
}

const PreviewPosterContent = ({ movie }: PreviewPosterContentProps) => {
    return (
        <div className={styles.container}>
            <div className={styles.container__icons}>
                <Image
                    className={[styles.container__icon, styles.container__icon_favorite].join(' ')}
                    src={bookmark}
                    alt="icon"
                />
                <Image
                    className={[styles.container__icon, styles.container__icon_similar].join(' ')}
                    src={wand}
                    alt="icon"
                />
                <Image
                    className={[styles.container__icon, styles.container__icon_rate].join(' ')}
                    src={star}
                    alt="icon"
                />
                <Image
                    className={[styles.container__icon, styles.container__icon_dislike].join(' ')}
                    src={circle}
                    alt="icon"
                />
            </div>

            <PreviewPosterContentBrief movie={movie as IMovie} className={styles.container__info} />
        </div>
    );
};

export default PreviewPosterContent;
/*<PreviewPosterContentBrief movie={movie as IMovie} className={styles.container__info} />*/
/*<div className={styles.container__info}>
                <div className={styles.container__rating}>
                    <span className={[styles.container__rating, styles.container__rating_integer].join(' ')}>
                        {`${movie.ratingKp}`.slice(0, 1)}
                    </span>
                    <span className={[styles.container__rating, styles.container__rating_fraction].join(' ')}>
                        {`${movie.ratingKp}`.slice(1, 3)}
                    </span>
                </div>
                <div className={styles.container__text}>
                    <p>
                        {movie.year}, {movie.countries[0].name}, {movie.genres[0].name}
                    </p>
                    <p>{movie.movieLength + ' ' + declineWord(movie.movieLength, ['минута', 'минуты', 'минут'])}</p>
                </div>
            </div>*/
