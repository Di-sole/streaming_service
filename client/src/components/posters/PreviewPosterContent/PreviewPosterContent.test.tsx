import { render, screen } from '@testing-library/react';
import PreviewPosterContent from './PreviewPosterContent';
import { movieTest } from '../../../testAsserts/testItems';

jest.mock('next-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

jest.mock('next/router', () => ({
    useRouter() {
        return {
            pathname: '',
        };
    },
}));

describe('PREVIEWPOSTERCONTENT SNAPSHOTS TESTS', () => {
    test('PreviewPosterContent should not be changed', () => {
        render(<PreviewPosterContent movie={movieTest} />);

        const id = screen.getByTestId('previewPosterContent');
        expect(id).toMatchSnapshot();
    });
});
