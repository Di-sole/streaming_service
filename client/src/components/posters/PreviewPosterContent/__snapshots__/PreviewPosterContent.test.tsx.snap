// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PREVIEWPOSTERCONTENT SNAPSHOTS TESTS PreviewPosterContent should not be changed 1`] = `
<div
  class="container"
  data-testid="previewPosterContent"
>
  <div
    class="container__icons"
  >
    <img
      alt="icon"
      class="container__icon container__icon_favorite"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      title="Смотреть позже"
      width="40"
    />
    <img
      alt="icon"
      class="container__icon container__icon_similar"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      title="Похожее"
      width="40"
    />
    <img
      alt="icon"
      class="container__icon container__icon_rate"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      title="Уже смотрел, оценить"
      width="40"
    />
    <img
      alt="icon"
      class="container__icon container__icon_dislike"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      title="Не нравится такое"
      width="40"
    />
  </div>
  <div
    class="container container__info"
    data-testid="previewPosterContentBrief"
  >
    <div
      class="container__rating"
    >
      <span
        class="container__rating container__rating_integer"
      >
        8
      </span>
      <span
        class="container__rating container__rating_fraction"
      >
        .8
      </span>
    </div>
    <div
      class="container__text"
    >
      <p>
        2011
        , 
        France
        ,
         
        Drama
      </p>
      <p>
        112 minutes.2
      </p>
    </div>
  </div>
</div>
`;
